src/Stan/Pattern/Type.hs

```diff
@@ -115,36 +116,45 @@ integerPattern =
 #endif
 
 -- | 'PatternType' for 'Natural'.
+naturalPattern :: PatternType
 naturalPattern =
 #if __GLASGOW_HASKELL__ < 900
   naturalPattern810
 #elif __GLASGOW_HASKELL__ >= 900
   naturalPattern900
 #endif
 
+#if __GLASGOW_HASKELL__ < 900
 integerPattern810 :: PatternType
 integerPattern810 = NameMeta
     { nameMetaName       = "Integer"
     , nameMetaModuleName = "GHC.Integer.Type"
     , nameMetaPackage    = "integer-wired-in"
     } |:: []
+#endif
 
+#if __GLASGOW_HASKELL__ >= 900
 integerPattern900 :: PatternType
 integerPattern900 = NameMeta
     { nameMetaName       = "Integer"
     , nameMetaModuleName = "GHC.Num.Integer"
     , nameMetaPackage    = "ghc-bignum"
     } |:: []
+#endif
 
+#if __GLASGOW_HASKELL__ < 900
 naturalPattern810 :: PatternType
 naturalPattern810 = "Natural" `baseNameFrom` "GHC.Natural" |:: []
+#endif
 
+#if __GLASGOW_HASKELL__ >= 900
 naturalPattern900 :: PatternType
 naturalPattern900 = NameMeta
     { nameMetaName       = "Natural"
     , nameMetaModuleName = "GHC.Num.Natural"
     , nameMetaPackage    = "ghc-bignum"
     } |:: []
+#endif
```

Comment:
Please undo the extra CPP here. I'd like to have a think whether there's a better way.

Reply:
@tomjaguarpaw, you may also not like it, but so you can see what it looks like 'on the page', I've put the CPP condition around everything affected by it - which also allows for some simplification within each limb of the condition.

Reply:
Yeah, I think that's the better way to go, but I'd rather get the slamdunks in first and then come back to the debatable cases later.

Reply:
OK.